spring-integration-xmpp/src/test/java/org/springframework/integration/xmpp/inbound/ChatMessageListeningEndpointTests.java

```diff
@@ -151,4 +167,110 @@ public void handleMessage(org.springframework.messaging.Message<?> message)
 		assertEquals("hello", ((MessagingException) msg.getPayload()).getFailedMessage().getPayload());
 	}
 
+	@Test
+	@SuppressWarnings("deprecation")
+	public void testExpression() throws Exception {
+		TestXMPPConnection testXMPPConnection = new TestXMPPConnection();
+
+		QueueChannel inputChannel = new QueueChannel();
+
+		ChatMessageListeningEndpoint endpoint = new ChatMessageListeningEndpoint(testXMPPConnection);
+		endpoint.setExtractPayload(false);
+		endpoint.setOutputChannel(inputChannel);
+		endpoint.setBeanFactory(mock(BeanFactory.class));
+		endpoint.afterPropertiesSet();
+		endpoint.start();
+
+		Message smackMessage = new Message();
+		smackMessage.setBody("foo");
+
+		XmlPullParser xmlPullParser = PacketParserUtils.newXmppParser(new StringReader(smackMessage.toString()));
+		xmlPullParser.next();
+		testXMPPConnection.parseAndProcessStanza(xmlPullParser);
+
+		org.springframework.messaging.Message<?> receive = inputChannel.receive(10000);
+		assertNotNull(receive);
+
+		Object payload = receive.getPayload();
+		assertThat(payload, instanceOf(Message.class));
+		assertEquals(smackMessage.getStanzaId(), ((Message) payload).getStanzaId());
+		assertEquals(smackMessage.getBody(), ((Message) payload).getBody());
+
+		Log logger = Mockito.spy(TestUtils.getPropertyValue(endpoint, "logger", Log.class));
+		given(logger.isInfoEnabled()).willReturn(true);
+
+		new DirectFieldAccessor(endpoint).setPropertyValue("logger", logger);
+
+		endpoint.setPayloadExpression(null);
+
+		smackMessage = new Message();
+		xmlPullParser = PacketParserUtils.newXmppParser(new StringReader(smackMessage.toString()));
+		xmlPullParser.next();
+		testXMPPConnection.parseAndProcessStanza(xmlPullParser);
+
+		ArgumentCaptor<String> argumentCaptor = new ArgumentCaptor<String>();
+
+		verify(logger).info(argumentCaptor.capture());
+
+
+		assertEquals("The XMPP Message [" + smackMessage + "] with empty body is ignored.",
+				argumentCaptor.getValue());
+
+		endpoint.stop();
+	}
+
+	@Test
+	public void testGcmExtension() throws Exception {
```

Comment:
That is outside of this task.
For the outbound part we have another JIRA: https://jira.spring.io/browse/INT-3541

Please, share your `sending` concerns there.

In this PR and its JIRA we discuss only the receiving part.